### PR TITLE
fix: while searching known users consume list of users instead of a flow

### PR DESCRIPTION
**Pull request by trOnk12**
…itting flow

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -187,6 +187,9 @@ dependencies {
     kaptAndroidTest(Libraries.Hilt.compiler)
     androidTestUtil(TestLibraries.testOrchestrator)
 
+    debugImplementation("androidx.compose.ui:ui-tooling:1.1.1")
+    implementation("androidx.compose.ui:ui-tooling-preview:1.1.1")
```
Comment (gongracr) [inline app/build.gradle.kts:191]: Shall we bump them to a Version variable?

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 2]
```diff
@@ -53,8 +54,8 @@ class NewConversationViewModel
     val state: SearchPeopleState by derivedStateOf {
         val noneSearchSucceed: Boolean =
             localContactSearchResult.searchResultState is SearchResultState.Failure &&
-                publicContactsSearchResult.searchResultState is SearchResultState.Failure &&
-                federatedContactSearchResult.searchResultState is SearchResultState.Failure
+                    publicContactsSearchResult.searchResultState is SearchResultState.Failure &&
+                    federatedContactSearchResult.searchResultState is SearchResultState.Failure
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:58]: Weird indentation?

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 3]
```diff
@@ -111,15 +112,18 @@ class NewConversationViewModel
...
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
-        }.catch {
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.Failure())
-        }.flowOn(Dispatchers.IO).collect {
-            localContactSearchResult = ContactSearchResult.InternalContact(
-                SearchResultState.Success(it.result.map { otherUser -> otherUser.toContact() })
-            )
+        localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
+
+        viewModelScope.launch {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:117]: do we need to launch this logic on a separate thread if even the function is defined as suspendable?

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt [hunk 4]
```diff
@@ -111,15 +112,18 @@ class NewConversationViewModel
...
+            localContactSearchResult = try {
+                val searchResult = searchKnownUsers(searchTerm)
+
+                ContactSearchResult.InternalContact(
+                    SearchResultState.Success(searchResult.result.map { otherUser -> otherUser.toContact() })
+                )
+            } catch (exception: Exception) {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt:124]: I think detekt will complain about having richer type of exceptions, so we should try to be more concrete or add a `Suppress` header for the method

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-589/1/) **failed**.

File: app/build.gradle.kts (hunk 1 above)
Comment (trOnk12) [reply app/build.gradle.kts:191]: ops didnt see it sneak in,sure will do

**Review by MohammedMokresh — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-589/4/) **failed**.

**Comment by trOnk12** [comment]
this fix was included in other PR

**Pull request closed without merging**